netlogo-core/src/main/agent/Box.scala

```diff
@@ -92,224 +93,33 @@ extends Topology(world2d, xWraps = false, yWraps = false) {
     else
       world.fastGetPatchAt(source.pxcor - 1, source.pycor + 1)
 
-  @throws(classOf[AgentException])
-  @throws(classOf[PatchException])
-  override def diffuse(amount: Double, vn: Int) {
-    val xx = world.worldWidth
-    val yy = world.worldHeight
-    val scratch = world.getPatchScratch
-    val scratch2 = Array.ofDim[Double](xx, yy)
-    val minx = world.minPxcor
-    val miny = world.minPycor
-    var x, y = 0
-
-    try while(y < yy) {
-      x = 0
-      while (x < xx) {
-        scratch(x)(y) =
-          world.fastGetPatchAt(x + minx, y + miny)
-            .getPatchVariable(vn)
-            .asInstanceOf[java.lang.Double].doubleValue
-        scratch2(x)(y) = 0
-        x += 1
+  override protected def diffuseCorners
+  (amount: Double, vn: Int, fourWay: Boolean, scratch: Array[Array[Double]]): Unit = {
```

Comment:
I'm not sure what idiomatic scala style is here, but the prevailing convention I have used in NetLogo always puts the open-paren immediately following the method name. Do you think this is better stylistically, or was this simply a typo?